libs/stats.js

```diff
@@ -91,6 +91,21 @@ module.exports = function(logger, portalConfig, poolConfigs){
         });
     }
 
+    // For block stats
+    this.getBlocksStats = function (cback) {
+        var client = redisClients[0].client;
+        client.hgetall("Allblocks", function (error, data) {
+            if (error) {
+                logger.log("error:-" + error);
+                cback("");
+                return;
+            }
+
+            cback(data);
+
+        });
+    };
+
```

Comment:
This results in the same JSON output without requiring the `Allblocks` data in redis and without querying redis on every request.
```
   this.getBlocks = function (cback) {
        var allBlocks = {};
        async.each(_this.stats.pools, function(pool, pcb) {

            for (var i=0; i<_this.stats.pools[pool.name].pending.blocks.length; i++)
                allBlocks[pool.name+"-"+_this.stats.pools[pool.name].pending.blocks[i].split(':')[2]] = _this.stats.pools[pool.name].pending.blocks[i];
        
            for (var i=0; i<_this.stats.pools[pool.name].confirmed.blocks.length; i++)
                allBlocks[pool.name+"-"+_this.stats.pools[pool.name].confirmed.blocks[i].split(':')[2]] = _this.stats.pools[pool.name].confirmed.blocks[i];
            
            pcb();
        }, function(err) {
            cback(allBlocks);            
        });
    };
```
Please give it a try.